src/gfn/gflownet/detailed_balance.py

```diff
@@ -27,17 +28,18 @@ class DBGFlowNet(PFBasedGFlowNet):
 
     def __init__(
         self,
+        pf: GFNModule,
+        pb: GFNModule,
         logF: ScalarEstimator,
+        on_policy: bool = False,
         forward_looking: bool = False,
-        **kwargs,
     ):
-        super().__init__(**kwargs)
+        super().__init__(pf, pb, on_policy=on_policy)
```

Comment:
I remember we used kwargs as a result of a last minute change while dual coding. In general, I'd rather have as few kwargs as possible. There is always a trade-off between:
- If we change something in the parent class init method, we need to do it for all subclasses
- The user might pass something in the init, thinking it will be processed as a kwarg, and it will silently fail.

Do you have a reason in mind for why we *need* kwargs?

Reply:
We certainly don't need them. I think the logic before (and the reason other projects do it, such as pytorch) is it's easier to maintain backward compatibility with API changes. However you're right that it makes the code more error prone from a user's point of view, it's harder to grok how the code works, and maybe it will make us think twice about changing APIs, so I'm happy to adopt it as a design philosophy going forward!